app/controllers/devise_token_auth/concerns/resource_finder.rb

```diff
@@ -12,14 +12,26 @@ def get_case_insensitive_field_from_resource_params(field)
     q_value
   end
 
-  def find_resource(field, value)
-    # fix for mysql default case insensitivity
-    q = "#{field.to_s} = ? AND provider='#{provider.to_s}'"
-    if ActiveRecord::Base.connection.adapter_name.downcase.starts_with? 'mysql'
-      q = "BINARY " + q
+  def find_resource
+
+    fields = (resource_params.keys.map(&:to_sym) & resource_class.authentication_keys)
+
+    conditions = []
+    values = {}
+    fields.each do |f|
+      q = " #{f.to_s} = :#{f.to_s} "
```

Comment:
use more meaningful variables, maybe `query_field` ?

Reply:
sure! just did it. called q = condition and f = field.